### PR TITLE
add default infer_probs

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Set attr.infer_probs = true as default for Proposal v4

### Tickets:
 - 82008

### CODE REVIEW

**Comment by iefode** [comment]
@ilyachur Could you please take a look?

File: src/core/src/op/proposal.cpp [hunk 1]
```diff
@@ -98,6 +98,7 @@ void op::v4::Proposal::validate_and_infer_types() {
     const auto& input0_type = get_input_element_type(0);
     set_output_type(0, input0_type, output_shapes[0]);
     set_output_type(1, input0_type, output_shapes[1]);
+    m_attrs.infer_probs = true;  // Proposal v4 requires default true of infer_probs so output_1 has valid data.
```
Comment (ilyachur) [inline src/core/src/op/proposal.cpp:101]: Does it fix some tests?

Comment (wilson-seok) [reply src/core/src/op/proposal.cpp:101]: @ilyachur Yes, it passes conformance_Proposal/ReadIRTest.ReadIR/PRC=f32_IR_name=Proposal_cf963b7ce3f0c56591caacf9b730b8192cf9e6dea6ef736bfd7a76587285aebf.xml_TargetDevice=TEMPLATE_Config=().

Comment (iefode) [reply src/core/src/op/proposal.cpp:101]: @ilyachur If you are ok with the changes, could you please merge it? :)

**Review by ilyachur — APPROVED** [review]

**Pull request merged**